### ISSUE TITLE
ComponentBuilder returns incorrect results

### ISSUE THREAD

**Issue by Nuubles** [2lstudios-mc/FlameCord#131]
Example:

```
BaseComponent[] c1 = TextComponent.fromLegacyText("§5Purple text");
BaseComponent[] c2 = TextComponent.fromLegacyText(" that continues in components");
ComponentBuilder builder = new ComponentBuilder();
for(BaseComponent c : c1)
  builder.append(c, FormatRetention.ALL); // or add
for(BaseComponent c : c2)
  builder.append(c, FormatRetention.ALL);
  
ProxiedPlayer player = ...;
player.sendMessage(builder.build()); // or something else that output BaseComponent[]
```

**Comment by Luccboy** [comment]
Hey, I couldn't reproduce your issue. I copied your code and the color from the first component is also applied to the second component:
![2022-05-21_17 39 08](https://user-images.githubusercontent.com/58391278/169658905-87382793-f79a-4426-9882-cbbb2424781c.png)
Here's my code:
![image](https://user-images.githubusercontent.com/58391278/169659088-f843dac1-55b6-4a49-9401-53def932fc36.png)
Which version of FlameCord and Spigot are you using?

**Comment by Nuubles** [comment]
I did a bit more testing and it seems that the problem I was having was related to the username issue ( https://github.com/2lstudios-mc/FlameCord/issues/130 ), and as it worked differently on both Bungee and Flame it resulted in the output being different as the colors "leaked". I've now replicated my issue on BungeeCord as well so I may have to redo some testing. I will close this issue as it seems to be more related to BungeeCord.

Thanks for testing!
